Source/DataModels/Scene/Scene.cpp

```diff
@@ -1363,3 +1417,96 @@ void Scene::ExecutePendingActions()
 		pendingCreateAndDeleteActions.pop();
 	}
 }
+
+std::vector<float> Scene::GetVertices()
+{
+	std::vector<float> result;
+	
+	for (ComponentMeshRenderer* meshRenderer : meshRenderers)
+	{
+		std::shared_ptr<ResourceMesh> mesh = meshRenderer->GetMesh();
+		GameObject* go = meshRenderer->GetOwner();
+		if (mesh != nullptr && go->CompareTag("NAVIGABLE"))
+		{
+			ComponentTransform* transform = go->GetComponent<ComponentTransform>();
+			//for (const ResourceMesh::Vertex& vertex : mesh->vertices)
+			//{
+			//	float4 transformedVertex = transform->GetGlobalMatrix() * float4(vertex.position, 1.0f);
+			for (const float3 vertex : mesh->GetVertices())
+			{
+				float4 transformedVertex = transform->GetGlobalMatrix() * float4(vertex, 1.0f);
+				result.push_back(transformedVertex.x);
+				result.push_back(transformedVertex.y);
+				result.push_back(transformedVertex.z);
+			}
+		}
+	}
+
+	return result;
+}
+
+std::vector<int> Scene::GetTriangles()
+{
+	int triangles = 0;
+	std::vector<int> maxVertMesh;
+	maxVertMesh.push_back(0);
+	for (ComponentMeshRenderer* meshRenderer : meshRenderers)
+	{
+		std::shared_ptr<ResourceMesh> mesh = meshRenderer->GetMesh();
+		if (mesh != nullptr && meshRenderer->GetOwner()->CompareTag("NAVIGABLE"))
+		{
+			//triangles += meshFaces.size() / 3;
+			triangles += mesh->GetFacesIndices().size();
+			maxVertMesh.push_back(mesh->GetVertices().size());
```

Comment:
```suggestion
			maxVertMesh.push_back(static_cast<int>(mesh->GetVertices().size()));
```
This raises a warning as well, try this.